### ISSUE TITLE
Couldn't query / find by nullable enum property

### ISSUE THREAD

**Issue by Deilan** [mbdavid/LiteDB#1051]
TL;DR: [Repository with the tests](https://github.com/deilan-issues-samples/LiteDB.Issues). More specifically:
- [LiteDatabaseNullableEnumPropertyTests](https://github.com/deilan-issues-samples/LiteDB.Issues/blob/master/LiteDB.Issues.Tests/LiteDatabase/LiteDatabaseNullableEnumPropertyTests.cs)
- [LiteRepositoryNullableEnumPropertyTests](https://github.com/deilan-issues-samples/LiteDB.Issues/blob/master/LiteDB.Issues.Tests/LiteRepository/LiteRepositoryNullableEnumPropertyTests.cs)

Model:
```csharp
public enum CustomerType
{
    Potential,
    New,
    Loyal
}
public class Customer
{
    public int Id { get; set; }
    public string Name { get; set; }
    public CustomerType? Type { get; set; }
}
```

Arrange:
```csharp
var col = _liteDatabase.GetCollection<Customer>();

var customer = new Customer
{
    Name = "John Doe",
    Type = CustomerType.New
};

col.Insert(customer);
```

Act:
```csharp
var customers = customersCollection
    .Find(Query.EQ(nameof(Customer.Type), CustomerType.New.ToString()))
    .ToArray();
```
or
```csharp
var customers = customersCollection
    .Find(x => x.Type == CustomerType.New)
    .ToArray();

```
or
```csharp
var customers = customersCollection
    .Find(x => x.Type.Value == CustomerType.New)
    .ToArray();
```
Assert:
```csharp
Assert.Single(customers.Where(c => c.Id == customer.Id));
```

Actually `customers` is empty. Fails for both `LiteDatabase` and `LiteRepository`.

**Comment by alb3ric** [comment]
Hello,

Is it plan to fix this issue or is there any workaround?

Thanks in advance!

Regards,

**Comment by mbdavid** [comment]
Hi guys,

I just fix v5 version to support Enum (and Enum nullable) as Integer value to support correct LINQ expressions (in LINQ, Enum as identified only as Integer value).

Take a look on this examples of how data is converted (only for v5)

LiteDB.Tests/Mapper/Enum_Tests.cs

**Comment by alb3ric** [comment]
Hello!

Great news! I'm going to take a look at it :)

Thanks

**Comment by JensSchadron** [comment]
I'm closing this issue as it has been marked as `fixed`
